velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -379,7 +419,16 @@ void HiveDataSink::appendData(RowVectorPtr input) {
   // must be zero.
   if (!isBucketed() && partitionIdGenerator_->numPartitions() == 1) {
     const auto index = ensureWriter(HiveWriterId{0});
-    write(index, input);
+    auto dataInput = input;
+    if (insertTableHandle_->tableStorageFormat() ==
+        dwio::common::FileFormat::PARQUET) {
```

Comment:
No need particular for PARQUET format. I think the change should apply to all formats we support so far.